### PR TITLE
[DATA-835] Add PackageService with Create/Delete/Get/ListPackage endpoints to API

### PR DESCRIPTION
**Pull request by tahiyasalam**

### CODE REVIEW

File: proto/viam/app/package/v1/package.proto
Comment (tahiyasalam) [inline proto/viam/app/package/v1/package.proto:28]: @agreenb wondering if comments, such as " // Primarily used by robot to download package files." from the scope doc should be included here

Comment (agreenb) [reply proto/viam/app/package/v1/package.proto:25]: Good question but nope, those were mostly used to describe the intent of each part of the scope.

File: proto/viam/app/package/v1/package.proto
Comment (agreenb) [inline proto/viam/app/package/v1/package.proto:25]: Since these all require authenticated requests, think can remove. Suggestion:

`// GetPackage returns the URL and metadata for a requested package version.`

File: proto/viam/app/package/v1/package.proto
Comment (agreenb) [inline proto/viam/app/package/v1/package.proto:11]: I think this can just be `// CreatePackage uploads a package to the cloud`

File: proto/viam/app/package/v1/package.proto
Comment (agreenb) [inline proto/viam/app/package/v1/package.proto:18]: `// DeletePackage removes the given package versions.`

File: proto/viam/app/package/v1/package.proto
Comment (agreenb) [inline proto/viam/app/package/v1/package.proto:95]: Can remove this comment and the one below.

File: proto/viam/app/package/v1/package.proto
Comment (agreenb) [inline proto/viam/app/package/v1/package.proto:32]: Something like: `// ListPackages get the URLs and metadata for requested packages. Include package name, version, and/or type to filter beyond the required organization_id.`

File: proto/viam/app/package/v1/package.proto
Comment (agreenb) [inline proto/viam/app/package/v1/package.proto:42]: Updated previously in scope to PACKAGE_TYPE_ARCHIVE based on feedback to represent a tar archive of files

File: proto/viam/app/package/v1/package.proto
Comment (agreenb) [inline proto/viam/app/package/v1/package.proto:63]: Can remove the limit from here since that'll be a variable that could be changed on the BE. I think ok to leave .tar.gz file comment in here since might be helpful just by looking at the API docs.

File: proto/viam/app/package/v1/package.proto
Comment (agreenb) [inline proto/viam/app/package/v1/package.proto:21]: Are these autogenerated? if not, think these {organization_id}'s can be removed here and below since organization is required but the methods involve a number of other required fields

**Review by agreenb — CHANGES_REQUESTED** [review]
Thanks for combining all of this into one and reflecting today's final changes to the doc!

**Comment by agreenb** [comment]
Mostly just nit suggestions for shorter gRPC method descriptions. Nice work!

File: proto/viam/app/package/v1/package.proto
Comment (tahiyasalam) [reply proto/viam/app/package/v1/package.proto]: done!

File: proto/viam/app/package/v1/package.proto
Comment (tahiyasalam) [reply proto/viam/app/package/v1/package.proto]: done!

File: proto/viam/app/package/v1/package.proto
Comment (tahiyasalam) [reply proto/viam/app/package/v1/package.proto]: done!

File: proto/viam/app/package/v1/package.proto
Comment (tahiyasalam) [reply proto/viam/app/package/v1/package.proto]: done!

File: proto/viam/app/package/v1/package.proto
Comment (tahiyasalam) [reply proto/viam/app/package/v1/package.proto]: done!

File: proto/viam/app/package/v1/package.proto
Comment (tahiyasalam) [reply proto/viam/app/package/v1/package.proto]: done! thanks for letting me know

File: proto/viam/app/package/v1/package.proto
Comment (tahiyasalam) [reply proto/viam/app/package/v1/package.proto]: done!

File: proto/viam/app/package/v1/package.proto
Comment (tahiyasalam) [reply proto/viam/app/package/v1/package.proto]: done~

File: proto/viam/app/package/v1/package.proto
Comment (agreenb) [inline proto/viam/app/package/v1/package.proto:14]: @sbal13 Would be useful to get your feedback on these names for consistency with typical REST endpoints. 

@tahiyasalam originally suggested :organization_id here but there will always be an org, package name, and package version, which seems like too much to include :)

File: proto/viam/app/package/v1/package.proto
Comment (agreenb) [inline proto/viam/app/package/v1/package.proto:35]: If the above remains v1/package, would switch this to v1/packages.

File: proto/viam/app/package/v1/package.proto
Comment (agreenb) [inline proto/viam/app/package/v1/package.proto:91]: `Package package = 1;`

File: proto/viam/app/package/v1/package.proto
Comment (agreenb) [inline proto/viam/app/package/v1/package.proto:84]: These will all be required and return a single package, whereas ListPackages will allow you to get multiple.

File: proto/viam/app/package/v1/package.proto
Comment (agreenb) [inline proto/viam/app/package/v1/package.proto:81]: [supernit] Not in the doc, but would switch over to URI for consistency with our other APIs (a superset of URL)

File: proto/viam/app/package/v1/package.proto
Comment (agreenb) [inline proto/viam/app/package/v1/package.proto:76]: [supernit] can put the comment above the field

File: proto/viam/app/package/v1/package.proto
Comment (agreenb) [inline proto/viam/app/package/v1/package.proto:63]: [supernit] Can put the comment above the field, since we'll update this comment in a follow-up PR to match the wording in [data_sync.proto](https://github.com/viamrobotics/api/pull/109#discussion_r1047683742)

**Review by agreenb — CHANGES_REQUESTED** [review]
Just a few last nit things, then looks great. Added @sbal13 to look at the REST apis

File: proto/viam/app/package/v1/package.proto
Comment (tahiyasalam) [reply proto/viam/app/package/v1/package.proto]: done

File: proto/viam/app/package/v1/package.proto
Comment (tahiyasalam) [reply proto/viam/app/package/v1/package.proto:86]: ah, right done

File: proto/viam/app/package/v1/package.proto
Comment (tahiyasalam) [reply proto/viam/app/package/v1/package.proto]: done

File: proto/viam/app/package/v1/package.proto
Comment (tahiyasalam) [reply proto/viam/app/package/v1/package.proto]: done

File: proto/viam/app/package/v1/package.proto
Comment (tahiyasalam) [reply proto/viam/app/package/v1/package.proto]: done

File: proto/viam/app/package/v1/package.proto
Comment (sbal13) [reply proto/viam/app/package/v1/package.proto:14]: Eric Adam and I talked about this and decided that the viam and app bits are unnecessary, though you'll see this format in other protos that were made before this. Basically we decided it should be `/<service>/<verison>/<action` so in this case just the `/package/v1/create` bit

Comment (sbal13) [reply proto/viam/app/package/v1/package.proto:14]: we're not necessarily going for REST conventions here, more just trying to make them expressive of the services/functions that these endpoints are proxies for

File: proto/viam/app/package/v1/package.proto
Comment (sbal13) [inline proto/viam/app/package/v1/package.proto:28]: nit: maybe `/package/v1/get`?

File: proto/viam/app/package/v1/package.proto
Comment (sbal13) [inline proto/viam/app/package/v1/package.proto:35]: nit: maybe `/package/v1/list`? open to whatever for these, just trying to stick to a convention where the last bit is a verb like 'create' or 'delete'

**Review by agreenb — COMMENTED** [review]
LGTM after Steven's suggestions. I think it'll require another re-approval after any commit so holding off until after. ✅✅

File: proto/viam/app/package/v1/package.proto
Comment (tahiyasalam) [reply proto/viam/app/package/v1/package.proto]: totally, makes sense!

File: proto/viam/app/package/v1/package.proto
Comment (tahiyasalam) [reply proto/viam/app/package/v1/package.proto]: done :) thanks!

File: proto/viam/app/package/v1/package.proto
Comment (tahiyasalam) [reply proto/viam/app/package/v1/package.proto:14]: cool!

Comment (agreenb) [reply proto/viam/app/package/v1/package.proto:14]: Thanks Steven - @tahiyasalam I think this all looks great except seems like can get rid of the /viam/app prefix!

Comment (tahiyasalam) [reply proto/viam/app/package/v1/package.proto]: my bad! should be fixed now

**Review by agreenb — APPROVED** [review]
🚀🚀🚀
Thanks for adding!

**Pull request merged**
